Source/FullScreenPokemon.d.ts

```diff
@@ -1931,6 +1938,11 @@ declare module FullScreenPokemon {
          * The partyActivate Function used to interact with this HMCharacter.
          */
         moveCallback: (player: IPlayer, pokemon: IPokemon) => void;
+
+        /**
+         * The name of the gym's leader.
```

Comment:
Badges are stored by the leader name in items.js so I thought that was the only way to see if I had them